From f34cfba178235ce8ed107e1fc3ccca7caec248af Mon Sep 17 00:00:00 2001
From: Gregory CLEMENT <gregory.clement@free-electrons.com>
Date: Wed, 25 May 2016 13:25:52 +0200
Subject: [PATCH 1739/2241] arm64: dts: marvell: Add xtal clock support for
 Armada 3700

The configuration of the clock depend of the gpio latch. This information
is stored in the gpio block registers. That's why the block is shared
using a syscon node.

Change-Id: I9abc3f368ebff5a108bc7ac7a9dc9df8e7959e3e
Signed-off-by: Gregory CLEMENT <gregory.clement@free-electrons.com>
Signed-off-by: allen yan <yanwei@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/40820
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Victor Gu <xigu@marvell.com>
---
 arch/arm64/boot/dts/marvell/armada-37xx.dtsi | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
index 196cb14..1e0b16f 100644
--- a/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-37xx.dtsi
@@ -296,7 +296,8 @@
 				status = "disabled";
 			};
 			gpio_nb: gpio@13800 {
-				compatible = "marvell,armada3700-gpio";
+				compatible = "marvell,armada3700-gpio",
+				"syscon", "simple-mfd";
 				reg = <0x13800 0x30
 					0x13c00 0x20>;
 				ngpios = <36>;
@@ -317,6 +318,12 @@
 					     <GIC_SPI 153 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 154 IRQ_TYPE_LEVEL_HIGH>,
 					     <GIC_SPI 155 IRQ_TYPE_LEVEL_HIGH>;
+
+				xtalclk: xtal-clk {
+					compatible = "marvell,armada-3700-xtal-clock";
+					clock-output-names = "xtal";
+					#clock-cells = <0>;
+				};
 			};
 
 			gpio_sb: gpio@18800 {
-- 
2.7.4

